proposals/0000-local-modules.rst

```diff
@@ -194,143 +195,294 @@ Motivation
    defined will be an error if used ambiguously.
 
-9. A new declaration form is introduced with the following BNF::
+   In other words, local modules are not entities of themselves: module
+   names are simply prefixes used in qualified names. This merging behavior
+   is thus a natural consequence of the overall design and agrees with
+   the behavior of standard Haskell (which allows multiple module aliases
+   to be the same).
+   
+#. A new declaration form is introduced with the following BNF::
 
      decl ::= ... | 'import' 'module' modid [ impspec ]
 
-   This declaration takes all entities (or those entities named in the
-   ``impspec``) in scope with a *modid*\ ``.`` and brings them into scope
-   into the local context (that is, the part of the program that other
-   declarations in the ``let`` or ``where`` scope over) without that
-   qualification. (They may be more deeply qualified, if *modid* exports local
-   modules.)
+   A. Without an ``impspec``: Consider all qualified names in scope with a
+      *modid*\ ``.`` qualification. These names are brought into scope (that is,
+      added to the local environment) without this *modid*\ ``.`` qualification.
+
+   #. With an ``impspec``: Each name included in the ``impspec`` must be in scope
+      with a *modid*\ ``.`` qualification. It is brought additionally into scope
+      without that qualification.
 
+      If a ``module N`` import item is included: Each name with a *modid*\ ``.N.``
+      qualification is brought into scope without the *modid*\ ``.`` qualification.
+      If an ``impspec`` is specified in a ``module N`` import item: apply this
+      approach recursively.
+     
    The declaration is allowed in ``let`` and ``where`` clauses.
 
    Note that the declaration form includes the word ``module`` to distinguish
    it from a normal ``import`` which induces a dependency on another file. An
    ``import module`` declaration cannot induce a dependency.
 
+   If the declaration brings no new identifiers into scope (because, for example,
+   the module name is not used to qualify any identifiers), it is an error.
+
    This declaration form is allowed only with ``-XLocalModules``.
 
-10. Local modules may be imported with this import item::
+#. Local modules may be exported with this export item::
+
+     export ::= ... | 'module' [ 'qualified' ] modid [ exports ]
+
+   The meaning of this export item depends on the presence of the ``qualified``
+   keyword in the export item.
+
+   A. With ``qualified``: This exports an export-module containing all names
+      in scope with the *modid*\ ``.`` qualification. If qualified names are in scope
+      further qualified by *modid*\ ``.``, export-modules for those further qualifications
+      are added to the export-module for *modid*. If an *exports* is specified, only those entities
+      are exported. Names in the *exports* can be written unqualified.
+
+      The new behavior is allowed only with ``-XLocalModules``.
```

Comment:
Move out of the sub-bullet point, probably. Since it applies to the entire syntax rather than just the `qualified` case, doesn't it?

Reply:
No. The other bullet is just restating current behavior.